Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,1539 @@
+
+    ScopedFileHandle fileResource("check_no_source_throws_test_file.hdf5");
+    auto destinationFile = fileResource.fullPath();
+
+    TS_ASSERT(compInfo.hasDetectorInfo()); // rule out throw by no detector info
+    TS_ASSERT(compInfo.hasSample());       // rule out throw by no sample
+    TS_ASSERT(!compInfo.hasSource());      // verify component has no source
+
+    TS_ASSERT_THROWS(NexusGeometrySave::saveInstrument(instr, destinationFile,
+                                                       DEFAULT_ROOT_PATH),
+                     std::invalid_argument &);
+  }
+
+  void test_sample_not_at_origin_throws() {
+
+    auto instrument = ComponentCreationHelper::createMinimalInstrument(
+        V3D(0, 0, -10), V3D(0, 0, 2), V3D(0, 0, 10));
+    auto instr = Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+
+    ScopedFileHandle fileResource("check_nxsource_group_test_file.hdf5");
+    std::string destinationFile = fileResource.fullPath();
+
+    TS_ASSERT_THROWS(NexusGeometrySave::saveInstrument(instr, destinationFile,
+                                                       DEFAULT_ROOT_PATH),
+                     std::invalid_argument &);
+  }
+
+  /*
+ ====================================================================
+
+ NEXUS FILE FORMAT TESTS
+
+ DESCRIPTION:
+
+ The following tests document that the file format produced by saveInstrument is
+ compliant to the present Nexus standard as of the date corresponding to the
+ latest version of this document.
+
+ LIST:
+
+ * test root group is NXentry class
+ * test NXinstrument group exists in root group
+ * test NXinstrument has expected name
+ * test NXsource group exists and is in NXinstrument group
+ * test NXsample group exists and is in NXentry group
+
+ ====================================================================
+ */
+
+  void test_root_group_is_nxentry_class() {
+
+    ScopedFileHandle fileResource("check_nxentry_group_test_file.nxs");
+    std::string destinationFile = fileResource.fullPath();
+
+    auto const &compInfo = (*m_instrument.first);
+    NexusGeometrySave::saveInstrument(m_instrument, destinationFile,
+                                      DEFAULT_ROOT_PATH);
+
+    HDF5FileTestUtility tester(destinationFile);
+    std::string dataSetName = compInfo.name(compInfo.root());
+
+    TS_ASSERT(tester.groupHasNxClass(NX_ENTRY, "/raw_data_1"));
+  }
+
+  void test_nxinstrument_group_exists_in_root_group() {
+
+    ScopedFileHandle fileResource("check_nxinstrument_group_test_file.hdf5");
+    std::string destinationFile = fileResource.fullPath();
+
+    auto const &compInfo = (*m_instrument.first);
+    const std::string expectedInstrumentName = compInfo.name(compInfo.root());
+
+    NexusGeometrySave::saveInstrument(m_instrument, destinationFile,
+                                      DEFAULT_ROOT_PATH);
+    HDF5FileTestUtility tester(destinationFile);
+    std::string dataSetName = compInfo.name(compInfo.root());
```

Comment:
Ok, can't see why you are extracting names as variables in all tests. Please remove from all following test methods.